### DISCUSSION TITLE
How to best provide reference genome for metaQUAST?

### DISCUSSION THREAD

**Discussion by Rob-murphys** [ablab/quast#166] [Q&A] [answered]
I ran metaQUAST without providing any references to allow it run its reference inference pipeline with the SILVA database, however it seems only three genomes were downloaded, of of them itself being a metagenome. Is there a better way to provide a reference for metagenome form an environmental sample in which we have no idea what to expect beyond broad strokes?

I ran with the following command:

`metaquast.py -o $outdir $assemblies`

My `assemblies` variable just contain a single multifasta of the assembled metagenome (done via `megahit`). Does submitting as one large multifasta effect the ability of metaQUAST?

best
Lamma

**Comment by alexeigurevich** [comment, marked as answer]
Hi Lamma,

> Is there a better way to provide a reference for metagenome form an environmental sample in which we have no idea what to expect beyond broad strokes?

If you have some clues about possible references you may either download them in fast format and provide them to metaQuast with `-r` option or create a sample txt file with species names and provided it with `--references-list` (see [FAQ Q10 in the manual](http://quast.sourceforge.net/docs/manual.html#faq_q10) for more details). In the latter case, metaQuast will try to search the species genomes in NCBI and download them.
For creating the list of species, you may try third-party tools to determine the species content of a metagenome assembly. Sorry, I can't recommend you state-of-the-art methods in this area since it is not my primary research topic. Note that using SILVA+BLAST inside metaQuast is a compromise between speed/accuracy/installation disk space. 

> Does submitting as one large multifasta effect the ability of metaQUAST?  

No, it should work approximately the same as if you submitted multiple shorter fasta files. The only difference is that for each input assembly we determine not more than X (50 by default) best hits in the SILVA database for further downloads. In your case, there are just 3 detected genomes, which is far lower than the top limit, so does not matter.

**Comment by Rob-murphys** [comment]
@alexeigurevich Thank for the reply.  I am mistaken and there is actually 28 references downloaded the metaQuast. This is still substantially below the diversity we know to exist in this microbiome though so that is curious!

I fear I may be using metaQuast wrong! Would it be possible for me to share the metaQuast output with you to check?

**Comment by alexeigurevich** [comment]
Your metaQuast running command is correct. It first runs BLAST search against SILVA and results are stored in the file like `<output_dir>quast_downloaded_references/blast.res_<your_assembly_name>` (you may find the exact path in `metaquast.log` in the line like "BLAST results for ... are saved to ..."). Normally, there are many hits there, the best ones are tried to be downloaded from NCBI. You may see in the log something like this "Trying to download found references from NCBI. Totally XX organisms to try."  This XX is the number of potentially good hits in SILVA. However, SILVA contains only 16S rRNA while the full genomes of the corresponding species may be absent on NCBI RefSeq or be of a very poor quality (too fragmented). In this case, the number of actually downloaded reference would be smaller than XX.

Potential options to proceed: 
If XX is 50, you may set a larger `--max-ref-number` value, e.g. 100 to try more SILVA hits in NCBI.
If you are using Quast v.5.0.2 (the latest stable release), you may try to switch to [v.5.1.0rc1](https://github.com/ablab/quast/releases/tag/quast_5.1.0rc1) (release candidate) which includes the latest SILVA database (v138), thus may find something missing in the previous version.

**Comment by Rob-murphys** [comment]
I tried the run again with `--max-ref-number`set to 100. I now get 59 references downloaded however in overall report I only see mention of two.

If i interpreting my report correctly it is indicating that 83% of the metagenome has aligned to these two references (which are actually the same species just different strain), however the kronoa chart shows only 50%? Either way this seems quite odd?

I cant share the full report here but have zipped it and shared it through OneDrive [here](https://alumni-my.sharepoint.com/:u:/g/personal/wmn262_ku_dk/EXUIMqGlj6BKps4JY_U1WAIB9Z2VlNpr8D3UfbuKHYWohA?e=eI4Yke)

**Comment by hugovalerio1** [comment]
Hi, Lamma
I had this same problem as you. As much as some time has passed, I believe this answer might be useful to someone. You can download reference genomes by following the following information from the NCBI website:
see item: 2. What is the easiest way to download data for multiple genome assemblies?
Available at: https://www.ncbi.nlm.nih.gov/genome/doc/ftpfaq/#GBorRS
